apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx

```diff
@@ -0,0 +1,199 @@
+import { VegaIcon, VegaIconNames } from '@vegaprotocol/ui-toolkit';
+import { useT } from '../../../lib/use-t';
+import classNames from 'classnames';
+import type { PartyVestingStats } from '@vegaprotocol/types';
+import BigNumber from 'bignumber.js';
+import { formatNumber } from '@vegaprotocol/utils';
+
+export const RewardHoarderBonus = ({
+  tiers,
+  vestingDetails,
+}: {
+  tiers:
+    | never[]
+    | {
+        minimum_quantum_balance: string;
+        reward_multiplier: string;
+      }[];
+  vestingDetails?: PartyVestingStats | null;
+}) => {
+  const t = useT();
+
+  const getUserTier = () => {
```

Comment:
I think this would be better moved into a hook for easier debugging

Reply:
moved in a hook for easier debugging